src/sql_generator.cpp

```diff
@@ -78,20 +80,26 @@ void create_schema(duckdb::Connection &con, const std::string &db_name,
 }
 
 void create_table(duckdb::Connection &con, const table_def &table,
-                  const std::vector<column_def> &columns) {
+                  const std::vector<const column_def *> &columns_pk,
+                  const std::vector<column_def> &all_columns) {
   const std::string absolute_table_name = table.to_string();
   std::ostringstream ddl;
   ddl << "CREATE OR REPLACE TABLE " << absolute_table_name << " (";
 
-  for (const auto &col : columns) {
+  for (const auto &col : all_columns) {
     ddl << KeywordHelper::WriteQuoted(col.name, '"') << " "
         << duckdb::EnumUtil::ToChars(col.type);
-    if (col.primary_key) {
-      ddl << " PRIMARY KEY";
-    }
     ddl << ", "; // DuckDB allows trailing commas
   }
 
+  if (!columns_pk.empty()) {
+    ddl << "PRIMARY KEY (";
+    write_joined(
+        ddl, columns_pk,
+        [](const std::string &name, std::ostringstream &out) { out << name; });
```

Comment:
do we need quoting ?

Reply:
great catch; made the `write_joined` method always quote the column name (since it knows for sure this is a column name) and refactored the trivial "just print the quoted column name" lambda out into a constant.